core/environment.go

```diff
@@ -114,9 +117,9 @@ func NewEnv(currentNs Symbol, stdin io.Reader, stdout io.Writer, stderr io.Write
 }
 
 func (env *Env) SetStdIO(stdin io.Reader, stdout io.Writer, stderr io.Writer) {
-	env.stdin.Value = &BufferedReader{bufio.NewReader(stdin)}
-	env.stdout.Value = &IOWriter{stdout}
-	env.stderr.Value = &IOWriter{stderr}
+	env.stdin.Value = MakeBufferedReader(stdin, stdinHashValue)
```

Comment:
So stdin, stdout and stderr will always have the same hash values regardless of the actual objects?

Reply:
Yes. At first this seemed strange, and I'm still on the fence about it, because:

- On the one hand, this is currently used only by the srepl() code, so there shouldn't be any conflict between this trio and what the repl() code would set up.
- Further, it might make sense (though I can't say why!) to have the same hash values for whatever comprises the "repl trio", regardless of what the repl uses for input.
- Yet, one could reasonably say that they're different file descriptors (which they are), so they should have different values.
- Clojure assigns them different values, but that makes sense since a Clojure sRepl is in addition to (instead of in place of) its standard repl. (In fact, it can handle multiple sRepls in a single instance). This might arguably be a good reason to keep the values the same for now, unless/until Joker also supports multiple active Repls per instance, so code could possibly detect that?

I can't think of any slam-dunk reasons to go with one or the other approach! Let me know whether you want me to use distinct hash values for these.

Reply:
Interestingly, while Joker assigns the same sRepl socket (from `l.Accept()`) to each of the three variables yet gives them differing hashes, Clojure gives (again, for an sRepl) `*in*` one hash while using the same hash for both `*out*` and `*err*`.

It's not clear to me what Joker should do here. Changing to be more like Clojure should probably be a distinct PR though.

Reply:
I think we should keep hash semantics clear and consistent with equality unless we have reasons to do otherwise. Two different (not equal) objects should have different hashes.

Reply:
That sounds good. I've updated the PR to better approximate what Clojure does, so the hash values for an sRepl are different from for a repl, still stable, and identical for `*out*` and `*err*`.